src/net_processing.cpp

```diff
@@ -2980,8 +2990,18 @@ void ProcessMessage(
         else if (state.GetResult() == TxValidationResult::TX_MISSING_INPUTS)
         {
             bool fRejectedParents = false; // It may be the case that the orphans parents have all been rejected
+
+            // Deduplicate parent txids, so that we don't have to loop over
+            // the same parent txid more than once down below.
+            std::vector<uint256> unique_parents;
+            unique_parents.reserve(tx.vin.size());
             for (const CTxIn& txin : tx.vin) {
-                if (recentRejects->contains(txin.prevout.hash)) {
+                unique_parents.push_back(txin.prevout.hash);
+            }
+            std::sort(unique_parents.begin(), unique_parents.end());
+            unique_parents.erase(std::unique(unique_parents.begin(), unique_parents.end()), unique_parents.end());
```

Comment:
nit: only if need to retouch, separating the `unique` operation from the `erase`, similar to the code example in https://en.cppreference.com/w/cpp/algorithm/unique, seems slightly clearer to me

```diff
-            unique_parents.erase(std::unique(unique_parents.begin(), unique_parents.end()), unique_parents.end());
+            auto last_unique_parent = std::unique(unique_parents.begin(), unique_parents.end());
+            unique_parents.erase(last_unique_parent, unique_parents.end());
```

Reply:
I disagree. The remove-erase (or in this case unique-erase) idiom is common in c++. Once you've seen it a couple of times it's clear what it's doing. Plus, unique leaves the last elements of the vector in indeterminate state, so we don't want anyone to be tempted to do anything with it between unique and erase.

Reply:
Fair.

Reply:
It's definitely clear. The only thing that would be a bit better is if instead of erasing, we just range over a Span(beigin, unique_it) because then we can do the erasing when the unique_parents vector is dropped. But I think that hash is trivially destructible, so not clear that erase will specialize with any overhead as there is no destructor to call. In any case, we only *actually* erase anything if there are duplicates, which I think is sufficiently rare of a case (but we still need to handle it correctly) that I'm not worried there would be much to erase.